### ISSUE TITLE
GCHP rundir script setCommonRunSettings.sh fails on "which python" if python is not installed on the node

### ISSUE THREAD

**Issue by yantosca** [geoschem/geos-chem#1762]
### Name and Institution (Required)

Name:  Bob Yantosca
Institution: Harvard + GCST

### Confirm you have reviewed the following documentation

- [x] [Support guidelines](https://geos-chem.readthedocs.io/en/stable/reference/SUPPORT.html)
- [x] [User manuals](https://geos-chem.readthedocs.io/en/stable/geos-chem-shared-docs/supplemental-guides/related-docs.html)
- [x] [Debugging GEOS-Chem and HEMCO errors](https://geos-chem.readthedocs.io/en/stable/geos-chem-shared-docs/supplemental-guides/debug-guide.html)
- [x] [Current and past Github issues](https://github.com/geoschem/geos-chem/issues)

### Description of your issue or question

I have been testing GCHP on new nodes that are set up with Rocky Linux.  These nodes do not have python installed in the usual locations:

```console
$ which python
/usr/bin/which: no python in (~/bin:~/bin/netcdf-scripts/scripts:/n/home09/ryantosca/miniconda3/condabin:~/bin:~/bin/netcdf-scripts/scripts:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/opt/puppetlabs/bin)
```

This causes the `setCommonRunSettings.sh` script to halt with an error.  The offending code is here:

https://github.com/geoschem/geos-chem/blob/4722f288e90291ba904222f4bbe4fc216d17c34a/run/GCHP/setCommonRunSettings.sh.template#L223-L237

This is because the `which python` returns failure before it even gets to the `$bc_ok` block.

I propose editing the if block as follows:
```bash
   # Use "bash calculator" if available; Python if not; fail otherwise
   which bc &> /dev/null; bc_ok=$?
   if [[ $bc_ok -eq 0 ]]; then
      # Use bash calculator
      SQRT=$(echo "sqrt (${Z})" | bc -l)
      N=$(echo $SQRT | awk '{print int($1+0.999)}')
   else
      which python &> /dev/null; py_ok=$?
      if [[ $py_ok -eq 0 ]]; then
	  # Use system Python
	  SQRT=$( python -c "import math; print(int(math.sqrt(${Z})))" )
	  N=$SQRT
      else
	  echo "Cannot auto-determine NX and NY (need either bc or python available)"
	  exit 70
      fi
   fi
```
This way we won't even have to test for python if the bash calculator is found. Bash calculator should come pre-installed (I think) with most Linux distros.

Tagging @lizziel @sdeastham

**Comment by yantosca** [comment]
I also think we might be able to use the bash `let` command syntax to obviate the need for bash calculator or python.  I will experiment a bit.

**Comment by lizziel** [comment]
Is this happening when you use `gchp.local.run`? I recently added `set -e` to that script which would be why it would be exiting the script early. 

A simple fix is to use `||`:

`which python &> /dev/null || py_ok=$?`

**Comment by lizziel** [comment]
Also, you'll need to set `py_ok` to 0 first. It'll be changed to 1 only if the which fails. We can put this update into the next Z version.

**Comment by yantosca** [comment]
@lizziel: Yes, this manifested with both gchp.local.run and gchp.run scripts. 

Yes, it might be the `-e`.  If we remove that we could trap errors after every relevant command.  

I think the || command would also fail since the `which python` fails and that gets trapped by `-e`.

**Comment by yantosca** [comment]
@lizziel: I could do a mockup of the logic in a bash script and let you know what works best.

**Comment by lizziel** [comment]
Using `||` should work since the return code of the line is from `||` rather than `which`.

**Comment by yantosca** [comment]
@lizziel: Feel free to close out this issue since it is now merged into dev/14.2.0.
